pkg/kube/krt/helpers.go

```diff
@@ -35,17 +36,27 @@ func getTypedKey[O any](a O) Key[O] {
 // GetKey returns the key for the provided object.
 // If there is none, this will panic.
 func GetKey[O any](a O) string {
-	if k, ok := tryGetKey[O](a); ok {
-		return k
+	as, ok := any(a).(string)
+	if ok {
+		return as
 	}
-
-	// Kubernetes types are pointers, which means our types would be double pointers
-	// Allow flattening
-	ao, ok := any(&a).(controllers.Object)
```

Comment:
This is the core change correct? The rest is just inlining the helper function

Reply:
Yep.